common/paramsbuilder/params.go

```diff
@@ -36,7 +46,7 @@ func (p *Client) ValidateParams() error {
 	return nil
 }
 
-func (p *Client) WithClient(
+func (p *Client) WithOauthClient(
```

Comment:
All connectors delegate to this parameter. This means all connectors will have either `WithOauthClient` or `WithBasicClient` method under the hood (more to come for sure) while universally surfacing `WithClient` method to whoever instantiates connector. Because of delegation the method can be renamed without problems.

Reply:
This doesn't seem strictly related to the pipeliner change, so I'd prefer to introduce this in a separate PR to get more feedback, especially since it touches a lot of files

Reply:
The Pipeliner is the first trigger, first reason for this change. 
But yes it is universal in a sense that more Auth Clients will be supported in future.
Here is a dedicated PR reflecting this goal: https://github.com/amp-labs/connectors/pull/712